scraper/src/strategies/default_strategy.py

```diff
@@ -187,6 +189,21 @@ def get_records_from_dom(self, current_page_url=None):
                 record['no_variables'] = record['url'] == record[
                     'url_without_variables']
 
+            # Define our own ObjectID to enable proper analytics
+            hierarchy_to_hash = {}
+            for lvl in hierarchy:
+                if hierarchy[lvl] is not None:
+                    hierarchy_to_hash[lvl] = hierarchy[lvl]
+            raw_hash = hashlib.sha1(json.dumps(
+                {'hierarchy_to_hash': json.dumps(hierarchy_to_hash,
+                                                 sort_keys=True),
+                 'url': record['url'],
+                 'position': position}, sort_keys=True).encode('utf-8'))
+            digest_hash = raw_hash.hexdigest()
+
+            # We only need 10 digits
+            record['objectID'] = int(digest_hash, 16) % 100000000000
```

Comment:
The modulo is reducing the entropy generated by the hash function.
If we absolutely need to limit the number of characters in the hash, let's use a hashing function whose digest can take a fixed length (i.e. shake or blake -- instead of sha1). If no limit is really mandatory on the size of objectID, let's keep the hash as is.

Reply:
Good point! Based on empirics checks, it seem that it should be an integer with 10 digits. I asked the internal support to see if they would promote a specific hash function and if there is some specific constrain on this value.

I will resolves this asap